### PR TITLE
Kpmp 1301 user portal service

### PR DESCRIPTION
**Pull request by zwright**

### CODE REVIEW

File: src/main/java/org/kpmp/auth/AuthController.java [hunk 1]
```diff
@@ -1,71 +1,75 @@
...
+            log.info(userFoundFmt, shibId, userAuth.toString());
+            return userAuth;
+        }
+        catch (HttpClientErrorException e) {
+            if (e.getStatusCode().equals(NOT_FOUND)) {
+                log.error(userNotFoundFmt, shibId);
+                throw new ResponseStatusException(HttpStatus.NOT_FOUND, "User " + shibId + " not found");
+            } else {
+                log.error(portalError + e.getStatusCode());
+                throw new ResponseStatusException(e.getStatusCode(), "There was a problem connecting to the User Portal");
```
Comment (rlreamy) [inline src/main/java/org/kpmp/auth/AuthController.java]: You have the portalError constant at the top, but you are not using it here.

File: src/main/java/org/kpmp/auth/AuthController.java [hunk 2]
```diff
@@ -1,71 +1,75 @@
...
+            log.info(userClientFoundFmt, clientId, shibId, userAuth.toString());
+            return userAuth;
+        }
+        catch (HttpClientErrorException e) {
+            if (e.getStatusCode().equals(NOT_FOUND)) {
+                log.error(userClientNotFoundFmt, clientId, shibId);
+                throw new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found");
+            } else {
+                log.error(portalError + e.getStatusCode());
+                throw new ResponseStatusException(e.getStatusCode(), "There was a problem connecting to the User Portal");
```
Comment (rlreamy) [inline src/main/java/org/kpmp/auth/AuthController.java]: You have the portalError constant at the top, but you are not using it here.

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,3 +1 @@
-server.port = 3030
-server.servlet.session.cookie.secure=true
-server.servlet.session.cookie.http-only=true
\ No newline at end of file
+userportal.url=https://testuserportal.kpmp.org
```
Comment (rlreamy) [inline src/main/resources/application.properties:1]: Have we added anything to a release card to make sure to switch this over to the production user portal?

File: src/test/java/org/kpmp/auth/AuthControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        userAuth.setShib_id("shibId");
+        when(userPortalService.getUserAuth("shibId")).thenReturn(userAuth);
+        assertEquals(authController.getUserInfo("shibId"), userAuth);
+        assertEquals(authController.getUserInfo("shibId").getShib_id(), "shibId");
+    }
+
+    @Test
+    public void testGetUserAuthNotFound() {
+        when(userPortalService.getUserAuth("shibId")).thenThrow(new HttpClientErrorException(HttpStatus.NOT_FOUND));
+        exception.expect(ResponseStatusException.class);
```
Comment (rlreamy) [inline src/test/java/org/kpmp/auth/AuthControllerTest.java:49]: I didn't know this was a thing you could do.  Neat!

File: src/test/java/org/kpmp/auth/UserAuthTest.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        testUserAuth = new UserAuth();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        testUserAuth = null;
+    }
+
+    @Test
+    public void testGettersAndSetters() throws Exception {
```
Comment (rlreamy) [inline src/test/java/org/kpmp/auth/UserAuthTest.java:24]: This is actually generally bad practice when writing tests.  Each test should be testing one piece of functionality.  The reason it is considered "bad practice" is as soon as one of the assertions fail, the test terminates and you don't know what else is broken.  I am not going to ask you to update this as part of the review, just pointing it out.

**Review by rlreamy — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 3 above)
Comment (zwright) [reply src/main/resources/application.properties:1]: https://docs.google.com/spreadsheets/d/1qohYQ6tb0KrwK6-pFo2LUBlPjZQ_cKHJN9gT79LXsnc

**Review by rlreamy — APPROVED** [review]

**Pull request merged**